### ISSUE TITLE
ConvertTo-Json for empty array produces null

### ISSUE THREAD

**Issue by smokedlinq** [PowerShell/PowerShell#23837]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

`$null -eq (@() | ConvertTo-Json -AsArray)` is `$true` but should be false because an empty array in json is `[]`

### Expected behavior

```console
PS> @() | ConvertTo-Json -AsArray
[]
```

### Actual behavior

```console
PS> @() | ConvertTo-Json -Array
```

### Error details

```console
N/A
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.2
PSEdition                      Core
GitCommitId                    7.4.2
OS                             Microsoft Windows 10.0.26120
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by smokedlinq** [comment]
I do realize this is because the pipeline has no items to send to the cmdlet, so the begin and end happen and no output, I would expect that the -AsArray with no items would be an empty json array `[]`

**Comment by smokedlinq** [comment]
If others want a workaround, use `ConvertTo-Json -InputObject @() -AsArray` but replace the `@()` with a variable since most likely that's what you are doing.

**Comment by mklement0** [comment]
`@()` sending _nothing_ to the pipeline is fundamental pipeline behavior (the _auto-enumeration_ of an empty enumerable results in no enumeration); `ConvertTo-Json` simply receives no input, and therefore outputs `$null` (technically, `[System.Management.Automation.Internal.AutomationNull]::Value`, which in expressions behaves like `$null`).

Indeed, `ConvertTo-Json -InputObject @()` is one workaround; in the pipeline, use:

```powershell
Write-Output -NoEnumerate @() | ConvertTo-Json

# More succinct, but obscure shortcut:
, @() | ConvertTo-Json
```

Note: 

* Do _not_ use `-AsArray` in these cases (passing an array _as a whole_ to `ConvertTo-Json`), because it'll create an _additional_ array that wraps the output array. 

* This behavior is problematic, as discussed in #10952

**Comment by roe-dk** [comment]
$Array | ConvertTo-Json will also flatten a single-item array so, in my experience, you are always better off using the -InputObject parameter
```
PS>@("OneValue") | ConvertTo-Json
"OneValue"
PS>ConvertTo-Json -InputObject @("OneValue")
[
  "OneValue"
]
PS>
```

**Comment by mklement0** [comment]
Good point, @roe-dk; since `ConvertTo-Json` of necessity collects all its pipeline input up front anyway before producing its single-string output, there's effectively no loss of _streaming_ functionality due to use of a direct argument (with `-InputObject`, which, btw, in `ConvertTo-Json`'s case is a _positional_ parameter, so that `ConvertTo-Json @("OneValue")` works too.

`-AsArray` is helpful if you have open-ended, streaming input from a cmdlet call, and you want to output a JSON array even if the cmdlet call, say `Get-ChildItem *.txt`, situationally only yields _one_ output object:

```powershell
# Ensures that a JSON *array* is created even if only ONE file matches.
# Two or more input objects of necessity always result in a JSON array.
Get-ChildItem *.txt | Select-Object Name | ConvertTo-Json -AsArray
```

However, as in the case at hand, there will be NO output if _no_ files matching `*.txt` happen to be present  (which, as noted, is signaled by the `[System.Management.Automation.Internal.AutomationNull]::Value` singleton as the output object, which presents as `$null` in expressions such as `$null -eq [System.Management.Automation.Internal.AutomationNull]::Value`) .

By contrast, if you use `@(...)`, the [array-subexpression operator](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_Operators#array-subexpression-operator--), with `-InputObject`, the no-input-object case yields `'[]'`

```powershell
# -> '[]'
ConvertTo-Json @(Get-ChildItem nosuch* | Select-Object Name )
```

Due to the behavior discussed in #10952, `-AsArray` mustn't be combined with input that already _is_ an array  (unless an extra array wrapper is desired), either due to passing an array directly to `-InputObject` or due to sending an array _as a whole_ through the pipeline (`Write-Output -NoEnumerate $array | ConvertTo-Json` or `, $array | ConvertTo-Json`).

**Comment by smokedlinq** [comment]
If we explicitly state we want it as an array using `-AsArray`, and the stream is empty, shouldn't we get an empty array?

**Comment by smokedlinq** [comment]
I understand and knew it's be a breaking behavior going in. Hoping for a 7.5 or 7.6 change 😍
